### PR TITLE
[home] Better type dev menu, fix bugs, add verified indicator

### PR DESCRIPTION
**Pull request by wschurman**
# Why

Instead of showing the warning in the CLI about it not being signed, we want to show it here in expo go.

PR that removed the warning in the CLI: https://github.com/expo/expo/pull/22975

Closes ENG-9056.

# How

A few steps were needed:
1. Add better types (this is very hard to guarantee correctness in its current state with `any` types)
2. Fix bugs in existing code surrounding manifest.
3. Add isVerified indicator. Open to design suggestions.

# Test Plan

Load an experience:

Screenshot 2023-06-23 at 9 04 21 AM

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by brentvatne** [comment]
@FiberJW - could you apply some 🌟 styling magic 🌟 to make this look a bit nicer?

**Comment by FiberJW** [comment]
Something like this?

Expo Go Dev Menu Unverified Indicator

**Comment by brentvatne** [comment]
that looks better! my only concern is how that works with longer app names. maybe we can keep it in the same spot and use that styling. also, we may want to make it yellow rather than red possibly. the only other concern i have here is that there is no context about what this actually means 😅 maybe if you press it, it'll use `Linking.openURL` to open up a website with more info?

**Comment by FiberJW** [comment]
Yeah, I was thinking that making it pressable would help people understand what this means.

Some more iterations:

Screenshot 2023-06-23 at 14 13 43
Screenshot 2023-06-23 at 14 13 38
Screenshot 2023-06-23 at 14 14 28

Screenshot 2023-06-23 at 14 13 30

**Comment by brentvatne** [comment]
i think i like the second one the best:
image

**Comment by wschurman** [comment]
(I couldn't find that icon in ionicons)

**Review by FiberJW — APPROVED** [review]
We can improve it further in a follow-up PR, but this looks good for now

File: home/menu/DevMenuTaskInfo.tsx [hunk 1]
```diff
@@ -1,45 +1,116 @@
...
+                color={theme.text.warning}
+                lightColor={theme.text.warning}
+                darkColor={theme.text.warning}
+                style={{
+                  marginHorizontal: 4,
+                }}
+              />
+              <Text
+                color="warning"
+                type="InterRegular"
```
Comment (FiberJW) [inline home/menu/DevMenuTaskInfo.tsx:105]: Might be nice to increase the weight of this text

```suggestion
                type="InterSemiBold"
```

**Comment by wschurman** [comment]
I'll add the expo.fyi link in a separate PR since it requires popping a browser over this drawer (if we keep in in-app) or linking out if we want it in a separate process, which seemed a bit out of the scope of this PR since popping a popover over an already-open popover is tough.

**Comment by brentvatne** [comment]
@wschurman - i think a separate process is fine, we can just use `Linking`

**Pull request merged**